CONTRIBUTING.md

```diff
@@ -167,6 +167,20 @@ https://help.github.com/en/github/getting-started-with-github/git-and-github-lea
 - Python [docstrings](http://www.python.org/dev/peps/pep-0257/) need to be in
   [reStructured Text (RST)](https://docutils.sourceforge.io/rst.html) format and
   compatible with [Sphinx](https://www.sphinx-doc.org).
+- String formatting should normally be done using
+  [formatted string literals (f-strings)](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals),
+  except:
+  - when the format string is kept in a separate variable, in which case the
+    [string ``format()`` method](https://docs.python.org/3/tutorial/inputoutput.html#the-string-format-method)
+    should be used;
+  - when [formatting a log message](https://docs.python.org/3/library/logging.html#logging.Logger.debug),
+    in which case the message format string (a
+    [`printf`-style](https://docs.python.org/3/library/stdtypes.html#old-string-formatting)
+    string literal) and its arguments should be passed separately to the logging
+    method.
+    This is a bit more efficient than using f-strings and allows for better log
+    aggregation, see e.g. this
```

Comment:
```suggestion
    aggregation, see, e.g., this
```
I believe we should add commas here. Although dropping "e.g."  altogether is another option (less formal, easier to read?)

Reply:
The commas around "e.g." are American English, see for example https://www.dailywritingtips.com/comma-after-i-e-and-e-g/ , but I agree that it's formal and better to avoid it.
I didn't use a straight "see this blog post" because it's not an authoritative source (just one that explains it well), so I am going to replace with "see for example this blog post", if that's OK.

Reply:
Yes, to me, "for example" seems like a good choice here. 

*(Regarding "e.g.": I remember from my grad school days that it's not followed by a comma in citations (in APA style, at least); but I wasn't sure about its usage in regular text)*

Reply:
I think 'e.g.' is totally fine to use and not significantly more formal than 'for example'; it's just shorthand.  The comma splice was the main thing I wanted adjusted, which we've left in here.

Reply:
I think the comma has been dropped: "see for example this blog post".

Reply:
I guess technically correct, but now we've got a comma splice without a comma, just two clauses shoved together that could use a spacer ;)

Reply:
If you prefer two sentences, that's fine with me.